### ISSUE TITLE
[Bug-App]: Spiking in batch logging when using WandbMetricsLogger

### ISSUE THREAD

**Issue by akum04** [wandb/wandb#9606]
### Describe the bug

I'm using  `wandb.integration.keras import WandbMetricsLogger` to log the loss values ,with `log_freq` set to `'batch'`. However, I encounter regular spikes at the end of epoch. 
To verify the values, I also logged using the `tf.keras.callbacks.TensorBoard` during the same training process, I didn't encounter such large spikes as in wandb. What is the difference and which one is actually calculating the loss values correctly across the batch.
No smoothing in the plots. no outliers excluded in both cases during plotting.

![loss_from_wandb](https://github.com/user-attachments/assets/9350783b-665d-419d-b31a-f8307c13fd7e)

![loss_from_tensorboard](https://github.com/user-attachments/assets/2d7b0cde-ac87-4b4b-9bda-dd38aadc25cd)

**Comment by ArtsiomWB** [comment]
Hey @akum04! Thank you for writing in!

Would love to look into this. What wandb version are you currently on, as well as could you please send us the example toy code snippet, so I could try reproducing this behavior on my end?

**Comment by akum04** [comment]
I'm using `wandb==0.18.0`. I'll try to create a seperate standalone code. 
But for now, I'm using the following callbacks in the `model.fit`.

For tensorboard callback:
`tensorboard_callback = tf.keras.callbacks.TensorBoard(
        log_dir=logdir, update_freq='batch')`

For wandb callback:
`wml = WandbMetricsLogger(log_freq='batch')`

Model.fit function 
`model.fit(train_dataset,
              epochs=cfg.training.num_epochs,
              batch_size=cfg.training.batch_size,
              steps_per_epoch=steps_per_epoch,
              callbacks=[checkpoint_callback, wml, tensorboard_callback],
              validation_data=val_dataset,
              validation_steps=validation_steps)`

**Comment by ArtsiomWB** [comment]
Hey @akum04! Any luck putting together the full script?

Unfortunately, I am not having much luck of reproducing this behavior on my end.

**Comment by ArtsiomWB** [comment]
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.
